app/src/main/java/org/wikipedia/page/PageFragmentLoadState.java

```diff
@@ -228,35 +228,29 @@ private void pageLoadFromNetwork(final ErrorCallback errorCallback) {
 
         app.getSessionFunnel().leadSectionFetchStart();
 
-        disposables.add(new PageClient()
-                .lead(model.getTitle().getWikiSite(), model.getCacheControl(), model.shouldSaveOffline() ? OfflineCacheInterceptor.SAVE_HEADER_SAVE : null,
-                        model.getCurEntry().getReferrer(), model.getTitle().getConvertedText(), calculateLeadImageWidth())
+
+        disposables.add(Observable.zip(new PageClient().summary(model.getTitle().getWikiSite(), model.getTitle().getPrefixedText(), null),
+                ServiceFactory.getRest(model.getTitle().getWikiSite()).getMediaList(model.getTitle().getConvertedText()), Pair::new)
                 .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread()).subscribe(rsp -> {
-                    app.getSessionFunnel().leadSectionFetchEnd();
-                    PageLead lead = rsp.body();
-                    pageLoadLeadSectionComplete(lead);
-
-                    bridge.execute(JavaScriptActionHandler.setFooter(fragment.requireContext(), model));
-
-                    if ((rsp.raw().cacheResponse() != null && rsp.raw().networkResponse() == null)
-                            || OfflineCacheInterceptor.SAVE_HEADER_SAVE.equals(rsp.headers().get(OfflineCacheInterceptor.SAVE_HEADER))) {
-                        showPageOfflineMessage(rsp.raw().header("date", ""));
-                    }
-                }, t -> {
-                    L.e("PageLead error: ", t);
-                    commonSectionFetchOnCatch(t);
-                }));
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(pair -> {
+                            createPage(pair.first.body(), pair.second);
+                            bridge.execute(JavaScriptActionHandler.setFooter(fragment.requireContext(), model));
+
+                            if ((pair.first.raw().cacheResponse() != null && pair.first.raw().networkResponse() == null)
+                                    || OfflineCacheInterceptor.SAVE_HEADER_SAVE.equals(pair.first.headers().get(OfflineCacheInterceptor.SAVE_HEADER))) {
+                                showPageOfflineMessage(Objects.requireNonNull(pair.first.raw().header("date", "")));
+                            }
+                        },
+                        throwable -> {
+                            L.e("Page details network response error: ", throwable);
+                            commonSectionFetchOnCatch(throwable);
+                        }));
```

Comment:
Looks like the page will break after clicking on an in-article link.

Reply:
@cooltey this should be fixed now... can you please confirm?

Reply:
Please see my other comment in `PageFragment.java`